File: src/main/java/com/EmosewaPixel/pixellib/materialSystem/types/TextureType.java
package com.EmosewaPixel.pixellib.materialsystem.types;

import com.EmosewaPixel.pixellib.materialsystem.lists.TextureTypes;

//Texture Types determine which sets of textures Material use for Object Types
public class TextureType {
    private String name;

    public TextureType(String name) {
        this.name = name;
        TextureTypes.add(this);
    }

    public String toString() {
        return name;
    }
}

File: src/main/java/com/EmosewaPixel/pixellib/materialSystem/types/TextureType.kt
package com.EmosewaPixel.pixellib.materialsystem.types

import com.EmosewaPixel.pixellib.materialsystem.lists.TextureTypes

//Texture Types determine which sets of textures Material use for Object Types
class TextureType(private val name: String) {
    init {
        TextureTypes.add(this)
    }

    override fun toString() = name
}